ocs_ci/ocs/managedservice.py

```diff
@@ -32,3 +42,51 @@ def get_dms_secret_name():
     return (
         config.DEPLOYMENT["addon_name"] + constants.MANAGED_DEADMANSSNITCH_SECRET_SUFFIX
     )
+
+
+def disable_odf_operator_and_update_pull_secret():
+    """
+    For unreleased odf operator version -  Disable odf operator
+      1. Disable odf operator
+      2. Update pull secrete
+      3. Create a catalogSource using the ocs-registry image
+
+    Note: This is a hack done to allow odf to odf deployment before full addon is available.
+    """
+    oc = ocp.OCP(kind=constants.SECRET, namespace="openshift-config")
+    logger.info("Disable odf operator")
+    exec_cmd(
+        'oc patch operatorhub.config.openshift.io/cluster -p=\'{"spec":{"sources":'
+        '[{"disabled":true,"name":"redhat-operators"}]}}\' --type=merge'
+    )
+    logger.info("Update pull secret")
+    pull_secret = exec_cmd("oc get -n openshift-config secret/pull-secret -o json")
+    pull_secret = oc.exec_oc_cmd("get -n openshift-config secret/pull-secret")
+    secret_data = pull_secret["data"][".dockerconfigjson"]
+    secret_data = base64.b64decode(secret_data).decode()
+    rhceph_dev_key = config["AUTH"]["quay-rhceph-dev-auth"]
+    secret_data.append(
+        f""""quay.io/rhceph-dev": \\{
+      "auth": "{rhceph_dev_key}",
+      "email": ""
+    \\}"""
+    )
+    with tempfile.NamedTemporaryFile() as secret_file:
+        secret_file.write(secret_data)
+        secret_file.flush()
+        exec_cmd(
+            f"oc set data secret/pull-secret -n openshift-config --from-file=.dockerconfigjson={secret_file.name}"
+        )
+
+    logger.info("Create a catalogSource using the ocs-registry image")
+    olm_data = request.get(
+        "http://perf1.perf.lab.eng.bos.redhat.com/shberry/odfodf/files/olm.yaml"
+    )
+    olm_data = list(yaml.safe_load_all(olm_data))
+    with tempfile.NamedTemporaryFile() as olm_file:
+        for olm_yaml in olm_data:
+            if olm_yaml.get("spec").get("image"):
+                olm_yaml["spec"]["image"] = "quay.io/rhceph-dev/ocs-registry:4.10.0-171"
```

Comment:
we need to change this . we cann't use constant value - 4.10.0-171 
best way is to get it from Jenkins parameter `OCS_REGISTRY_IMAGE` and
 default to stable version if not specified `quay.io/rhceph-dev/ocs-registry:latest-stable-4.10`